src/main/java/com/aws/greengrass/componentmanager/KernelConfigResolver.java

```diff
@@ -505,22 +543,22 @@ private String replace(String stringValue, ComponentIdentifier componentIdentifi
                 continue;
             }
 
-            if (!resolvedKernelServiceConfig.containsKey(targetComponent)) {
+            if (!Objects.isNull(resolvedKernelServiceConfig)
+                    && !resolvedKernelServiceConfig.containsKey(targetComponent)) {
                 LOGGER.atError().kv("interpolation text", matcher.group()).kv("target component", targetComponent)
                         .kv("main component", componentIdentifier.getName())
                         .log("Failed to interpolate because the target component is not in resolved Nucleus services."
                                      + " This indicates the dependency resolution is broken.");
                 continue;
             }
 
-            if (CONFIGURATION_NAMESPACE.equals(namespace)) {
+            if (shouldInterpolateConfiguration && CONFIGURATION_NAMESPACE.equals(namespace)) {
                 Optional<String> configReplacement =
                         lookupConfigurationValueForComponent(targetComponent, key, resolvedKernelServiceConfig);
                 if (configReplacement.isPresent()) {
                     stringValue = stringValue.replace(matcher.group(), configReplacement.get());
                 }
-
-            } else if (systemParameters.containsKey(namespace)) {
+            } else if (!Objects.isNull(resolvedKernelServiceConfig) && systemParameters.containsKey(namespace)) {
```

Comment:
Why is this needed now?

Reply:
Just do the validation before calling. Is there any concern to do so?

Reply:
Well it is a new check which was not considered necessary before. If the value is null it seems that we have some massive problems. I think we would want it to fail here.

Reply:
Can the value ever be null?

Reply:
With our current code, not likely. The `resolvedKernelServiceConfig` was initialized at the very beginning.

But `resolve` is a public method, in the future it's possible that someone pass an empty one.

Reply:
If prefer that you undo this.